packages/gatsby/src/bootstrap/load-themes/index.js

```diff
@@ -5,6 +5,34 @@ const _ = require(`lodash`)
 const debug = require(`debug`)(`gatsby:load-themes`)
 const preferDefault = require(`../prefer-default`)
 const getConfigFile = require(`../get-config-file`)
+const {
+  validatePluginOptions,
+  formatOptionsError,
+} = require(`../../utils/validate-plugin-options`)
+const reporter = require(`gatsby-cli/lib/reporter`)
+const Joi = require(`@hapi/joi`)
+
+const getPluginOptions = theme => {
+  const options = theme.options || {}
+  try {
+    const { validatePluginOptions: validateAPI } = require(path.join(
+      theme.themeDir,
+      `gatsby-node.js`
+    ))
+    // note: this is fairly naive; it doesn't pass the additional API options
+    // that other APIs have available (and e.g. that a plugin implementing this API) would have
```

Comment:
This is sort of the big reason I kinda hate it. I don't think there are many other valid/useful API options -- but I don't like the "gotcha" that this kind of API is special for themes and is treated differently than e.g. a plugin.